src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -257,6 +257,13 @@
       + ", and decrease " + BATCH_SIZE_CONFIG + " configuration properties.";
   private static final String WRITE_METHOD_DISPLAY = "Write Method";
 
+  public static final String RETRY_ON_CONFLICT_CONFIG = "retry.on.conflict";
+  private static final String RETRY_ON_CONFLICT_DISPLAY = "Retry on conflict";
+  private static final String RETRY_ON_CONFLICT_DOC = "Specify how many times the operation should"
+      + " be retried by Elasticsearch, when write conflicts occur, while using the write method "
+      + WriteMethod.UPSERT.toString() + ". The value is the number of retries.";
+  private static final int RETRY_ON_CONFLICT_DEFAULT = 0;
```

Comment:
Was there any reason why this default was chosen? 

I found this q&a https://discuss.elastic.co/t/whats-appropriate-value-at-retry-on-conflict/54312 which says 

> If you need parallel indexing of similar documents, what are the worst case outcomes. 

So I'm wondering if this means an appropriate default would be proportional to the number of tasks that are set

Reply:
Since each task is a thread, I think parallel indexing would be of the order of numOfTasks (maybe there is more detail as to how parallel indexing happens within the connector and how it is split) 

While we can add a new config to make this configurable going forward, we might be able to backport a fix to set retry_on_conflict(numOfTasks) without the use of an extra config. WDYT? 

Can we run a manual test/IT with parallel update requests to see if that is a possible solution?

Reply:
That approach would work, for `numOfTasks` the property we have available is `tasks.max`, is that the prop you had in mind?

Reply:
Yes, that's right 👍

Reply:
@levzem what do you think about the above approach? Do you think users would need their own config of this value based on the [raised issue](https://github.com/confluentinc/kafka-connect-elasticsearch/issues/374)? Should we go for the backported default value in this PR and issue a new config later if requested?

Reply:
ideally we shouldn't even be producing write conflicts and not have to introduce this config. otherwise, I would like to implement this without the config at all as I don't think this adds value to the customer and adds unnecessary complexity. the Jest client rework will remove the need for this config, so let's not add it in the first place